src/Block/Fetch.hs

```diff
@@ -291,22 +291,10 @@ fetchNextBlock = do
         $ Text.pack $ "Error decoding RequestNext response: " <> e
     Right (RollBackward _point _tip) ->
       logWarnNS "fetchNextBlock" "Received RollBackward response"
-    Right (RollForward (UnsupportedBlock type_ raw) _tip) ->
-      logWarnNS
-        "fetchNextBlock"
-        $ "Received unsupported block in the RollForward response with type: "
-          <> type_
-          <> " raw: " -- TODO: raw output only on debug logging level
-          <> raw
-    Right (RollForward (MkNoDatumBlock type_ block) _tip) -> do
-      let block' = noDatum2datumBlock block
-      liftIO $ atomically $ writeTBQueue env.queue block'
-      logInfoNS "fetchNextBlock" $
-        "Fetched no datum " <> type_ <> " block: " <> Text.pack (show (block.slot, block.hash))
-    Right (RollForward (MkDatumBlock type_ block) _tip) -> do
+    Right (RollForward block _tip) -> do
```

Comment:
Maybe we need a more explicit error message in case we can't decode it upper?

Reply:
Yes, but if I don't mistake it will be a message like: `Left "Error in $: not enough input"` without any additional information, like the original message?